jgrapht-core/src/test/java/org/jgrapht/generate/PruferTreeGeneratorTest.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * (C) Copyright 2018-2018, by Alexandru Valeanu and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.generate;
+
+import org.jgrapht.Graph;
+import org.jgrapht.GraphTests;
+import org.jgrapht.graph.DefaultEdge;
+import org.jgrapht.graph.DirectedAcyclicGraph;
+import org.jgrapht.graph.SimpleGraph;
+import org.jgrapht.util.SupplierUtil;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Random;
+
+/**
+ * Tests for {@link PruferTreeGenerator}
+ *
+ * @author Alexandru Valeanu
+ */
+public class PruferTreeGeneratorTest {
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testNullPruferSequence(){
+        Graph<Integer, DefaultEdge> tree = new SimpleGraph<>(SupplierUtil.createIntegerSupplier(1),
+                SupplierUtil.DEFAULT_EDGE_SUPPLIER, false);
+
+        PruferTreeGenerator<Integer, DefaultEdge> generator =
+                new PruferTreeGenerator<>(null);
+    }
+
+    @Test
+    public void testEmptyPruferSequence(){
+        Graph<Integer, DefaultEdge> tree = new SimpleGraph<>(SupplierUtil.createIntegerSupplier(1),
+                SupplierUtil.DEFAULT_EDGE_SUPPLIER, false);
+
+        PruferTreeGenerator<Integer, DefaultEdge> generator =
+                new PruferTreeGenerator<>(new int[]{});
+
+        generator.generateGraph(tree);
+        Assert.assertEquals(2, tree.vertexSet().size());
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testInvalidPruferSequence(){
+        Graph<Integer, DefaultEdge> tree = new SimpleGraph<>(SupplierUtil.createIntegerSupplier(1),
+                SupplierUtil.DEFAULT_EDGE_SUPPLIER, false);
+
+        PruferTreeGenerator<Integer, DefaultEdge> generator =
+                new PruferTreeGenerator<>(new int[]{10});
+    }
+
+    @Test
+    public void testPruferSequence(){
+        Graph<Integer, DefaultEdge> tree = new SimpleGraph<>(SupplierUtil.createIntegerSupplier(1),
+                SupplierUtil.DEFAULT_EDGE_SUPPLIER, false);
+
+        PruferTreeGenerator<Integer, DefaultEdge> generator =
+                new PruferTreeGenerator<>(new int[]{4, 4, 4, 5});
+
```

Comment:
Can you verify the result structure?

Reply:
Not really. I would need to verify that the generated tree is isomorphic to the one I used to create the Prufer code. And that PR is not yet merged. I did a simpler test, for the degree sequences.

Reply:
Got it.  You could use VF2, but yeah, better to just follow up with your tree isomorphism.